uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/directives/leave-request-popup.js

```diff
@@ -34,12 +34,18 @@ define([
           contactId: '<',
           leaveRequest: '<',
           leaveType: '@',
-          forStaff: '<'
+          userRole: '@'
         },
         restrict: 'EA',
         link: function (scope, element) {
           var controller = _.capitalize(getLeaveType(scope.leaveType, scope.leaveRequest)) + 'RequestCtrl';
 
+          if (!scope.userRole) {
+            $log.error("User role not specified.");
```

Comment:
this is pointless, either you throw an error or you fallback to a default. Given that you are falling back to a default, you can simply do

```js
directiveOptions: function () {
  return {
    userRole: scope.userRole || 'staff';
  }
});